### DISCUSSION TITLE
Helm release failure reason not found in helm release status/description

### DISCUSSION THREAD

**Discussion by soumyabk** [fluxcd/flux2#4203] [Q&A] [answered]
If the helmrelease reconciliation fails because some resources is not ready(for instance, all the pods of the daemonsets are not ready), we don't see any information about what resource was pending that caused the helmrelease status=failed. 
Is there a way we can see the cause of failure? 

 
      Helm upgrade failed: timed out waiting for the condition
      Last Helm logs:

      resetting values to the chart's original version
      performing update for monitoring
      creating upgraded release for monitoring
      waiting for release monitoring resources (created: 0 updated: 231  deleted: 1)
      warning: Upgrade "monitoring" failed: timed out waiting for the condition

**Comment by soumyabk** [comment]
version information:


    flux: v0.28.5
    helm-controller: v0.32.2
    kustomize-controller: v1.0.0-rc.2
    source-controller: v1.0.0-rc.2

**Comment by soumyabk** [comment]
for instance in the example[ here](https://fluxcd.io/flux/cheatsheets/troubleshooting/#how-to-debug-install-retries-exhausted-errors), it shows that the deployment is not ready, but i dont see that in the helmrelease description

**Comment by soumyabk** [comment]
@kingdonb do you have an pointers or recommendations we could try

**Reply by kingdonb** [reply, marked as answer]
> flux: v0.28.5

It looks as though you're running a very old flux CLI. Be sure to use `brew upgrade flux` if you installed via homebrew, you might want to also upgrade the controllers in your cluster. You should use the latest version.

I suspect you've picked older versions because you are on an older version of Kubernetes. You must be on at least 1.20.6 to avoid server-side apply bugs that existed in earlier versions, but according to [the supported matrix](https://fluxcd.io/flux/installation/#prerequisites) you should be on a supported version of Kubernetes to receive support. If you find a problem in a version between 1.20.6 and 1.27 we will still be interested (I haven't seen any issues, personally), but we cannot support old versions of Flux or Kubernetes before GA because they can't be fixed any longer. You will need to use a supported release.

I think the output has been improved since 0.28, looking at `flux get helmreleases` typically does not show failure status information beyond the "install/upgrade retries exhausted" message when Helm failed, but you can read those details in an event emitted by the helmrelease for a limited time. Kubernetes events are expired and deleted from the server after something like an hour. So if it failed some time ago, there might be no direct logged evidence remaining of why it failed. You can run the upgrade again by suspending and resuming the HelmRelease, or by rolling back whatever upgrade failed using the Helm cli.

**Comment by soumyabk** [comment]
yes `flux get helmreleases` does not show failure information, but looking at the docs, `kubectl describe helmrease` should give information about what resources the helmreleases was waiting on before it timed out(also i am running these checks right after it times out). 
We're currently running https://github.com/fluxcd/flux2/releases/tag/v2.0.0-rc.2 for all the controllers. 
The client version should not affect the results we see from `kubectl describe` so im wondering if there is any flag or anything we need to enable to be able to get that information about a failing helmrelease. Without that it is very hard to debug what is causing the helmrelease to fail.

**Reply by kingdonb** [reply]
The CLI version affects more your ability to read the output. In your earlier reply, it looks like you need to upgrade your CLI.

Can you confirm the CLI is upgraded? The `kubectl describe helmrelease` output will only show you associated events. There are a number of reasons those events could be occluded, and only some of them are a bug. Can you try posting your HelmRelease YAML

Also, please use the threading feature.
